control-plane/agents/src/bin/core/controller/io_engine/v1/translation.rs

```diff
@@ -593,6 +594,86 @@ impl AgentToIoEngine for transport::ImportPool {
     }
 }
 
+impl AgentToIoEngine for GetRebuildRecord {
+    type IoEngineMessage = v1::nexus::RebuildHistoryRequest;
+    fn to_rpc(&self) -> Self::IoEngineMessage {
+        v1::nexus::RebuildHistoryRequest {
+            uuid: self.nexus.clone().to_string(),
+        }
+    }
+}
+
+impl TryIoEngineToAgent for v1::nexus::RebuildHistoryResponse {
+    type AgentMessage = transport::RebuildHistory;
+    /// This converts gRPC pool object into Control plane Pool state.
+    fn try_to_agent(&self) -> Result<Self::AgentMessage, SvcError> {
+        Ok(Self::AgentMessage {
+            uuid: NexusId::try_from(self.nexus.as_str()).map_err(|_| SvcError::InvalidUuid {
+                uuid: self.nexus.to_owned(),
+                kind: ResourceKind::Nexus,
+            })?,
+            records: self
+                .records
+                .iter()
+                .map(|record| (ExternalType(record.clone())).try_into())
+                .collect::<Result<Vec<_>, _>>()?,
+        })
+    }
+}
+
+impl TryFrom<ExternalType<v1::nexus::RebuildHistoryRecord>> for RebuildRecord {
+    type Error = SvcError;
+    fn try_from(value: ExternalType<v1::nexus::RebuildHistoryRecord>) -> Result<Self, Self::Error> {
+        Ok(Self {
+            child_uri: ChildUri::try_from(value.0.child_uri.as_str()).map_err(|_| {
```

Comment:
Since you're moving value, I think you just do  move values here to, eg:
ChildUri::try_from(value.0.child_uri)

Reply:
Im using child_uri inside closure right, To form SvcError. Hence cant move the value

Reply:
Ah I had missed that, you're returning that in the error, yeah let's keep it then, it's good for debugging if it ever goes wrong.